tests/unit/components/drawer/ComponentDetailPanel.spec.js

```diff
@@ -107,6 +109,44 @@ describe('test component: Plugin Component Detail Panel', () => {
     });
   });
 
+  describe('Test function: getUnreferencedAttributes', () => {
+    it('should return an existing attribute', () => {
+      const attributes = { name: 'attribute', definition: null };
+      const component = {
+        attributes: [attributes],
+      };
+
+      expect(wrapper.vm.getUnreferencedAttributes(component)).toEqual([{
+        name: 'attribute',
+        definition: null,
+      }]);
+    });
+  });
+
+  describe('Test function: getSelectedComponentAttributes', () => {
+    it('should return referenced attributes', () => {
+      const refAttributes = { name: 'refAttribute', definition: { definedAttributes: [] } };
+      const unrefAttributes = { name: 'unrefAttribute', definition: null };
+      wrapper.vm.selectedComponentAttributes = [refAttributes, unrefAttributes];
+
+      expect(wrapper.vm.getSelectedComponentAttributes('referenced')).toEqual([refAttributes]);
+    });
+
+    it('should return unreferenced attributes', () => {
+      const refAttributes = { name: 'refAttribute', definition: { definedAttributes: [] } };
+      const unrefAttributes = { name: 'unrefAttribute', definition: null };
+      wrapper.vm.selectedComponentAttributes = [refAttributes, unrefAttributes];
+
+      expect(wrapper.vm.getSelectedComponentAttributes('unreferenced')).toEqual([unrefAttributes]);
+    });
+
+    it('should return empty array', () => {
+      wrapper.vm.selectedComponentAttributes = [];
```

Comment:
put something inside to show that the key is not recognized.

Reply:
done